activesupport/lib/active_support/cache.rb

```diff
@@ -18,10 +18,6 @@ module Cache
     autoload :NullStore,        "active_support/cache/null_store"
     autoload :RedisCacheStore,  "active_support/cache/redis_cache_store"
 
-    # These options mean something to all cache implementations. Individual cache
-    # implementations may support additional options.
-    UNIVERSAL_OPTIONS = [:namespace, :compress, :compress_threshold, :expires_in, :expire_in, :expired_in, :race_condition_ttl, :coder, :skip_nil]
```

Comment:
We can't change the name and location of this constant without deprecation, it is in the public API https://edgeapi.rubyonrails.org/classes/ActiveSupport/Cache.html

We could move it and assign UNIVERSAL_OPTIONS to the new one, and then deprecate this one.

Reply:
Thanks for the pointer 👍 I assume this would go on a separate PR?
I did a quick grep and couldn't see any examples of constants being deprecated. 
How could we wrap accessing the constant with a deprecation warning?

Reply:
I think you should do it in this PR. You could use [`deprecate_constant`](https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/activesupport/lib/active_support/deprecation/constant_accessor.rb#L5).

Reply:
Alejandro, I don't think this PR needs to do anything with this constant.

It's defined in this base module with the intention that all caches can have predictable options. It happens to currently only be used in the subclass `MemCacheStore` as part of its initialization, but that's an implementation detail. (I get that it's a little confusing, and if I were writing it from scratch, I would write it differently too.)

I don't think this constant should be deprecated, or renamed, or moved, and even if I'm wrong, I think this constant isn't related to the purpose of this PR. If we want to change something about the way any of the four cache subclasses pick their options for getting initialized, I think it's reasonable to open that in a separate issue/PR.